stellargraph/layer/knowledge_graph.py

```diff
@@ -670,6 +671,200 @@ def __init__(
         )
 
 
+class RotHEScoring(Layer, KGScore):
+    def __init__(self, hyperbolic):
+        self._hyperbolic = hyperbolic
+        if self._hyperbolic:
+            self._convert = lambda c, v: poincare_ball_exp(c, None, v)
+            self._add = poincare_ball_mobius_add
+            self._squared_distance = lambda c, v, w: tf.square(
+                poincare_ball_distance(c, v, w)
+            )
+        else:
+            self._convert = lambda _c, v: v
+            self._add = lambda _c, v, w: v + w
+            self._squared_distance = lambda _c, v, w: tf.reduce_sum(
+                tf.math.squared_difference(v, w), axis=-1
+            )
```

Comment:
Should we define `curvature_prime` in init?
```suggestion
            )
            self.curvature_prime = None
```

Reply:
Our other model classes don't stub out their weights (e.g. GraphConvolution's `kernel` and `bias` etc.), and it seems Keras itself doesn't either, e.g. `Dense`: https://github.com/tensorflow/tensorflow/blob/1248bc9c61dd4a1e6677c9cc1c747b792765ae58/tensorflow/python/keras/layers/core.py#L1127-L1186 . I guess this could be a first 😄  Thoughts?

Reply:
Ah in that case, I'm happy for you to just leave it. I guess it's probably more important for member variables that are supposed to be accessed from outside the class.